yarn-project/pxe/src/synchronizer/synchronizer.ts

```diff
@@ -13,14 +17,30 @@ import { type PxeDatabase } from '../database/index.js';
  * details, and fetch transactions by hash. The Synchronizer ensures that it maintains the note processors
  * in sync with the blockchain while handling retries and errors gracefully.
  */
-export class Synchronizer {
-  private runningPromise?: RunningPromise;
+export class Synchronizer implements L2BlockStreamEventHandler {
   private running = false;
   private initialSyncBlockNumber = INITIAL_L2_BLOCK_NUM - 1;
   private log: DebugLogger;
+  private blockStream: L2BlockStream;
 
-  constructor(private node: AztecNode, private db: PxeDatabase, private jobQueue: SerialQueue, logSuffix = '') {
+  constructor(private node: AztecNode, private db: PxeDatabase, private l2TipsStore: L2TipsStore, logSuffix = '') {
     this.log = createDebugLogger(logSuffix ? `aztec:pxe_synchronizer_${logSuffix}` : 'aztec:pxe_synchronizer');
+    this.blockStream = new L2BlockStream(node, this.l2TipsStore, this);
```

Comment:
This object seems to get built in `PXEService` in `pxe_service.ts`, but that file did not get updated. Maybe build errors were not caught due to the optional `logSuffix` param?

Reply:
Possibly!